### ISSUE TITLE
Errors when loading modelica:// referenced CAD files when running Simulate with Animation

### ISSUE THREAD

**Issue by rehlm-gse** [OpenModelica/OpenModelica#11067]
I can confirm that I am still having issues loading STL/DXF files when using `modelica://` referenced paths similar to #9503 when running OpenModelica v1.22.0-dev-263-gfae4e4871e (64-bit) on Windows 10. 

I have tested this by trying to run the Modelica.Mechanics.MultiBody.Examples.Systems.RobotR3.FullRobot model with animation. When doing so, I get the following error type for all 7 CAD shapes that are in the model:

[1] 10:13:58 Scripting Error
Could not find the file .

[8] 16:01:50 Error
Failed to lookup URI (is the package loaded?) modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b0.dxf

In the Modelica.Mechanics.MultiBody.Examples.Systems.RobotR3.Utilities.MechanicalStructure submodel that is referenced by the FullRobot example, the custom shapeType for the above file is specified as follows:

```
Modelica.Mechanics.MultiBody.Parts.BodyShape b0(
    r={0,0.351,0},
    shapeType="modelica://Modelica/Resources/Data/Shapes/RobotR3/b0.dxf",
    r_shape={0,0,0},
    lengthDirection={1,0,0},
    widthDirection={0,1,0},
    length=0.225,
    width=0.3,
    height=0.3,
    color={0,0,255},
    animation=animation,
    animateSphere=false,
    r_CM={0,0,0},
    m=1)
```

It looks like this gets correctly translated to an absolute file path and stored in the FullRobot_visual.xml file except for the fact that the `modelica://` part doesn't get converted to `file://`:
```
<shape>
      <ident>mechanics.b0.shape1</ident>
      <type>modelica://C:/development/modelica/RobotR3/Resources/Data/Shapes/RobotR3/b0.dxf</type>
      <T>
...
</shape>
```
 When the xml file is parsed by Visualization.cpp it sees the `modelica://` in the absolute file path and tries to convert it again which is when the error gets raised.

**Comment by adeas31** [comment]
I get,

```
<ident>mechanics.b0.shape1</ident>
<type>modelica://Modelica/Resources/Data/Shapes/RobotR3/b0.dxf</type>
```

which is the expected behavior.

**Comment by rehlm-gse** [comment]
After applying your first patch I get the following output:

```
check 
cylinder 
failed 
---------- 
check 
cylinder 
failed 
---------- 
check 
cylinder 
failed 
---------- 
check 
cylinder 
failed 
---------- 
check 
cylinder 
failed 
---------- 
check 
cylinder 
failed 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b0.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b0.dxf 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b1.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b1.dxf 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b2.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b2.dxf 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b3.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b3.dxf 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b4.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b4.dxf 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b5.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b5.dxf 
---------- 
check 
modelica://Modelica/Resources/Data/Shapes/RobotR3/b6.dxf 
succeeded 
modelica://C:/development/.openmodelica/libraries/Modelica 4.0.0+maint.om/Resources/Data/Shapes/RobotR3/b6.dxf 
---------- 
check 
cylinder 
failed 
---------- 
check 
sphere 
failed 
---------- 
```

**Comment by rehlm-gse** [comment]
The second diff gives this output:

```
SCST shape = cylinder 
SCST shape = cylinder 
SCST shape = cylinder 
SCST shape = cylinder 
SCST shape = cylinder 
SCST shape = cylinder 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b0.dxf 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b1.dxf 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b2.dxf 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b3.dxf 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b4.dxf 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b5.dxf 
SCST shape = modelica://Modelica/Resources/Data/Shapes/RobotR3/b6.dxf 
SCST shape = cylinder 
SCST shape = sphere 
```

**Comment by rehlm-gse** [comment]
Your question on whether I am sure that I am running MSL 4.0.0 is a bit of an interesting one. Here are the annotations for the MSL library that I have loaded (came with one of my OpenModelica installs):

```
preferredView="info",
version="4.0.0",
versionDate="2020-06-04",
dateModified = "2020-06-04 11:00:00Z",
revisionId="74d056ef1 2022-10-07 11:39:57 +0000",
```

You'll notice that the git commit hash and date don't match the dateModified or the versionDate annotations. If I download MSL 4.0.0 from Github and look up the annotations for the fresh download, all the annotations do roughly match up for that one:

```
preferredView="info",
version="4.0.0",
versionDate="2020-06-04",
dateModified = "2020-06-04 11:00:00Z",
revisionId="6626538a2 2020-06-04 19:56:34 +0200",
```

The `74d056ef1` commit SHA isn't one that exists in the Modelica/ModelicaStandardLibrary repo, so I'm not sure where it came from. There are a few differences between the two libraries, but nothing to note in the `FullRobot` example or the CAD resources that it loads.

As a double check, I retargeted my OMEdit environment at the MSL 4.0.0 library that I directly downloaded from Github and am still seeing the same behavior. 

The update that you suggested fixes the problem though, so I am good for now. Thank you for the quick support on this.

Ryan
